SignalServiceKit/src/Util/OWSAnalytics.m

```diff
@@ -157,6 +157,12 @@ - (void)tryToSyncEvents
 
         DDLogDebug(@"%@ trying to deliver event: %@", self.tag, firstEventKey);
         self.hasRequestInFlight = YES;
+
+        __block UIBackgroundTaskIdentifier task;
```

Comment:
This goes back to what we were discussing this morning.  Objective-C captures by value by default, Swift captures by reference.  `task` isn't set yet when the block is defined, so we can't capture it's value.  `__block` has two side effects: it causes this variable to be captured by reference and allocates it on the heap instead of the stack so that it outlives the scope in which its defined.

![screen shot 2017-07-24 at 5 15 24 pm](https://user-images.githubusercontent.com/625803/28545112-b644294e-7093-11e7-8dd3-6dc5ca68e1b9.png)